src/algorithm/minkowski/gjk/mod.rs

```diff
@@ -115,6 +117,115 @@ where
         None
     }
 
+    /// Do time of impact intersection testing on the given primitives, and return a valid collision
+    /// time of impact.
+    ///
+    /// ## Parameters:
+    ///
+    /// - `left`: left primitive
+    /// - `left_transform`: model-to-world-transform for the left primitive
+    /// - `right`: right primitive,
+    /// - `right_transform`: model-to-world-transform for the right primitive
+    ///
+    /// ## Returns:
+    ///
+    /// Will optionally return the time of impact. If no collision was detected, None is returned.
+    #[allow(unused_variables)]
+    pub fn intersect_time_of_impact<P, PL, PR, TL, TR>(
+        &self,
+        left: &PL,
+        left_start_transform: &TL,
+        left_end_transform: &TL,
+        right: &PR,
+        right_start_transform: &TR,
+        right_end_transform: &TR,
+    ) -> Option<P::Scalar>
+    where
+        P: EuclideanSpace,
+        P::Scalar: BaseFloat,
+        PL: SupportFunction<Point = P>,
+        PR: SupportFunction<Point = P>,
+        SP: SimplexProcessor<Point = P>,
+        P::Diff: Neg<Output = P::Diff> + InnerSpace,
+        TL: Transform<P> + TranslationInterpolate<P::Scalar>,
```

Comment:
I'm curious to see the compile times on that :)

Reply:
They're not that bad tbh :)